expr/func_movingwindow.go

```diff
@@ -0,0 +1,167 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/grafana/metrictank/batch"
+	"github.com/grafana/metrictank/errors"
+	"github.com/grafana/metrictank/schema"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/consolidation"
+	"github.com/raintank/dur"
+)
+
+type FuncMovingWindow struct {
+	in           GraphiteFunc
+	windowSize   string
+	fn           string
+	xFilesFactor float64
+
+	shiftOffset uint32
+}
+
+// NewMovingWindowConstructor takes an agg string and returns a constructor function
+func NewMovingWindowConstructor(name string) func() GraphiteFunc {
+	return func() GraphiteFunc {
+		return &FuncMovingWindow{fn: name}
+	}
+}
+
+func NewMovingWindow() GraphiteFunc {
+	return &FuncMovingWindow{fn: "average", xFilesFactor: 0}
+}
+
+func (s *FuncMovingWindow) Signature() ([]Arg, []Arg) {
+	return []Arg{
+			ArgSeriesList{val: &s.in},
+			ArgString{key: "windowSize", val: &s.windowSize, validator: []Validator{IsSignedIntervalString}},
```

Comment:
incompatibilities with graphite should be documented in docs/graphite.md (@robert-milan please keep this in mind)

Reply:
Should I mark it `stable` in docs/graphite.md? Just clarifying.

Reply:
so, the fact that we only allow a window size as interval string, not as a integer number of points, is an exception to document.
if, that aside, everything else behaves exactly like graphite does - to your knowledge - then yes you can mark it as stable.